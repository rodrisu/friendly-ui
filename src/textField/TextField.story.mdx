import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks'

import { BASIC } from '../_utils/taxonomy'
import { inputModes, inputTypes, TextField } from './index'

<Meta title={`${BASIC}/Text Inputs`} />

# **TextField**

<Canvas>
  <Story name="Inputs (Text Field)">
    <TextField
      type={inputTypes.TEXT}
      id="id"
      name="inputName"
      inputMode={inputModes.TEXT}
      pattern="pattern"
      placeholder="Placeholder"
      labelledBy="aria label"
      disabled={false}
      readOnly={false}
      label=""
      error=""
      onChange={() => {}}
      onFocus={() => {}}
      onBlur={() => {}}
      focus={false}
      required={false}
      maxLength={1000}
      autoComplete="off"
      title="accessibility text"
    />
  </Story>
</Canvas>

## Specifications

> Empty unfocus state

In unfocus state the placholder is displayed.

> Empty focus state

When user taps on the input we display the cursor and the keyboard and we keep the placeholder as long as the user doesn’t start typing.

> Typing focus state

When user starts typing we hide the placeholder and (if that’s the case) display the cross on the right that allows the user to erase the input.

When user taps on the cross we move back to the empty focus state (placeholder + cursor + keyboard).

When input has help icon it doesn’t have a cross. Help icon opens a new modal screen with related explanations.

> Filled unfocus state

When user taps outside of the input area, starts scrolling or hide the keyboard we unfocus the input (hide the cursor) but we keep the cross (if that’s the case) and the value entered. If this value is too long when unfocus we display the beginning of the value.
When user re-taps on the input we place the cursor after the last character typed.

> Behaviour

If users come back to a previous step in which there was an input, we remember what the user input in the field (and so display the next button for example when needed).

> Long texts

We can’t go on 2 lines, if text is too long we just don’t show the beginning of the text.

> Read only input field.

Input fields can be read only, which means they display user provided data, but you can’t directly type into them.

When tapped the open another screen dedicated for providing the input or
a system picker.

They don’t have a focused state and don’t contain the clear button.

> Focus state

Active input field should have a 2pt blue inner border.

## Usage

```jsx
import { TextField } from '@blablacar/ui-library/build/textField'
;<TextField />
```

<ArgsTable of={TextField} />
