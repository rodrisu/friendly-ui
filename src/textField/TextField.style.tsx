import styled from 'styled-components'

import { color, font, inputBorderSize, radius, space, transition } from '../_utils/branding'
import { normalizeHorizontally } from '../layout/layoutNormalizer'

export const StyledTextField = styled.div`
  position: relative;
  box-sizing: border-box;
  ${normalizeHorizontally};
  padding: ${space.m} 0;

  & .kirk-textField-wrapper {
    position: relative;
    display: flex;
    box-sizing: border-box;
    color: ${color.midnightGreen};
    background-color: ${color.lightGray};
    border-radius: ${radius.l};
    border: solid ${inputBorderSize.default} ${color.lightGray};
    box-shadow: none;
  }

  & .kirk-textField-wrapper--hasFocus {
    border: ${inputBorderSize.focus} solid ${color.blue};
  }

  & .kirk-textField-wrapper--hasFocus input,
  & .kirk-textField-wrapper--hasFocus textarea {
    padding: calc(${space.l} + ${inputBorderSize.default} - ${inputBorderSize.focus});
  }

  & .kirk-textField-wrapper--hasFocus input {
    padding-right: 0;
  }

  & input {
    appearance: none;
    border: 0;
    border-radius: ${radius.l};
    background-color: ${color.lightGray};
    color: ${color.midnightGreen};
    flex: 1;
    font-size: ${font.base.size};
    line-height: ${font.base.lineHeight};
    width: 100%;
    caret-color: ${color.blue};
    padding: ${space.l};
  }

  & input {
    padding-right: 0;
    /* Make sure to override the user agent default font-family on input */
    font-family: inherit;
    /*
    Use margin-right instead of padding-right to fix a cursor bug
    when calling setSelectionRange on Safari Mobile [BBCSPA-1030]
    */
    margin-right: 48px;
  }

  & input:not(:first-child) {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
    padding-left: 0;
  }

  & input::placeholder {
    color: ${color.lightMidnightGreen};
  }

  & input::-ms-clear {
    display: none;
  }

  & input[type='number']::-webkit-outer-spin-button,
  & input[type='number']::-webkit-inner-spin-button {
    -webkit-appearance: none;
  }

  & input[type='number'] {
    -moz-appearance: textfield;
    -webkit-appearance: none;
  }

  & input[type='search'] {
    width: 100%;
    box-sizing: border-box;
  }

  & input[type='number'],
  & input[type='search'] {
    box-shadow: none;
  }

  & input:focus {
    border-color: ${color.lightGray};
    box-shadow: none;
    outline: none;
  }

  &.kirk-error .kirk-textField-wrapper {
    background: ${color.lightRed};
    border: solid 1px ${color.lightRed};
    animation: textFieldError ${transition.duration.fast} ease-in-out;
  }

  &.kirk-error input {
    background-color: ${color.lightRed};
  }

  &.kirk-error .kirk-error-message {
    color: ${color.red};
    display: block;
    padding: ${space.m};
  }

  & label {
    color: ${color.midnightGreen};
    padding: 0 ${space.m} ${space.s} ${space.m};
  }

  & .kirk-button {
    background-color: transparent;
    color: ${color.lightMidnightGreen};
    height: auto;
  }

  & .kirk-textField-button {
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
  }

  & .kirk-textField-button[hidden] {
    /* we use visiblity hidden instead of display none to resolve a display glitch
    on Safari Mobile when clearing input value and hiding button simultaneously */
    visibility: hidden;
  }

  &.kirk-disabled {
    opacity: 0.5;
    cursor: default;
  }

  @keyframes textFieldError {
    20% {
      margin-left: -10px;
      margin-right: 10px;
    }
    40% {
      margin-left: 10px;
      margin-right: -10px;
    }
    60% {
      margin-left: -5px;
      margin-right: 5px;
    }
    80% {
      margin-left: 5px;
      margin-right: -5px;
    }
    100% {
      margin-left: 0px;
      margin-right: 0px;
    }
  }
`
