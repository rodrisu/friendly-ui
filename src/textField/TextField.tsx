import React, { InputHTMLAttributes, PureComponent } from 'react'
import cc from 'classcat'
import isEmpty from 'lodash.isempty'

import { prefix } from '../_utils'
import { color } from '../_utils/branding'
import { OnChangeParameters } from '../_utils/onChange'
import { Button, ButtonStatus } from '../button'
import { CrossIcon } from '../icon/crossIcon'
import { EyeIcon } from '../icon/eyeIcon'
import { StyledTextField } from './TextField.style'

export type textfield = HTMLInputElement | HTMLTextAreaElement

export enum inputTypes {
  TEXT = 'text',
  PASSWORD = 'password',
  EMAIL = 'email',
  NUMBER = 'number',
  SEARCH = 'search',
  TEL = 'tel',
}

export enum inputModes {
  NONE = 'none',
  TEXT = 'text',
  DECIMAL = 'decimal',
  NUMERIC = 'numeric',
  TEL = 'tel',
  SEARCH = 'search',
  EMAIL = 'email',
  URL = 'url',
}

export type CommonFormFields = Readonly<{
  name: string
  id?: string
  type?: inputTypes
  placeholder?: string
  maxLength?: number
  autoCorrect?: 'on' | 'off'
  autoComplete?: string
  spellCheck?: 'true' | 'false'
  disabled?: boolean
  readOnly?: boolean
  autoFocus?: boolean
  required?: boolean
  title?: string
  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void
}>

type errorField = string | JSX.Element

export type TextFieldProps = CommonFormFields &
  Readonly<{
    defaultValue?: string
    labelledBy?: string
    onChange?: (obj: OnChangeParameters) => void
    onClear?: () => void
    className?: string
    errorClassName?: string
    error?: errorField
    addon?: JSX.Element
    label?: string
    buttonTitle?: string
    focus?: boolean
    inputMode?: inputModes
    pattern?: string
    inputRef?: (input: textfield) => void
    format?: (value: string, previousValue: string) => string
    focusBorder?: boolean
    loader?: JSX.Element
    inputAttributes: InputHTMLAttributes<HTMLInputElement>
  }>

export type TextFieldState = {
  value: string
  previousValue: string
  defaultValue: string
  showPassword: boolean
  hasFocus: boolean
}

export class TextField extends PureComponent<TextFieldProps, TextFieldState> {
  private input: HTMLInputElement | HTMLTextAreaElement
  static defaultProps: Partial<TextFieldProps> = {
    inputRef() {},
    onClear() {},
    onFocus() {},
    onBlur() {},
    type: inputTypes.TEXT,
    format: value => value,
    focusBorder: true,
  }

  static INPUT_TYPES = inputTypes
  static INPUT_MODES = inputModes

  state = {
    value: this.props.defaultValue,
    defaultValue: this.props.defaultValue,
    previousValue: '',
    showPassword: false,
    hasFocus: false,
  }

  clearButton: HTMLButtonElement = null

  componentDidMount() {
    if (this.input && this.props.focus) {
      this.input.focus()
    }
  }

  static getDerivedStateFromProps(props: TextFieldProps, state: TextFieldState) {
    if (props.defaultValue !== state.defaultValue) {
      return {
        ...state,
        value: props.defaultValue,
        defaultValue: props.defaultValue,
        previousValue: state.value,
      }
    }
    return null
  }

  componentDidUpdate(prevProps: TextFieldProps) {
    if (this.props.focus && this.props.focus !== prevProps.focus) {
      this.input.focus()
    }
  }

  onTextFieldChange = (event: React.ChangeEvent<textfield>) => {
    const { value } = this.state
    this.setState(
      {
        value: this.props.format(event.target.value, value),
        previousValue: value,
      },
      this.onChange,
    )

    if (event.currentTarget.value === '') {
      this.props.onClear()
    }
  }

  onChange = () => {
    this.props.onChange({
      name: this.props.name,
      value: this.state.value,
    })
  }

  onFocus = (event: React.FocusEvent<HTMLInputElement>) => {
    this.setState({
      hasFocus: true,
    })
    this.props.onFocus(event)
  }

  onBlur = (event: React.FocusEvent<HTMLInputElement>) => {
    if (!event.relatedTarget || event.relatedTarget !== this.clearButton) {
      this.setState({
        hasFocus: false,
      })
      this.props.onBlur(event)
    }
  }

  clearValue = () => {
    const { value } = this.state
    this.setState(
      {
        value: '',
        previousValue: value,
      },
      () => {
        this.input.focus()
        this.onChange()
        this.props.onClear()
      },
    )
  }

  toggleShowPassword = () => {
    this.setState(({ showPassword }) => {
      this.input.focus()
      return { showPassword: !showPassword }
    })
  }

  ref = (input: textfield) => {
    this.input = input
    this.props.inputRef(input)
  }

  renderError = () => {
    const { error, errorClassName } = this.props
    const className = cc(['kirk-error-message', errorClassName])

    return React.isValidElement(error) ? (
      React.cloneElement(error, { className } as Object)
    ) : (
      <span role="alert" className={className}>
        {error}
      </span>
    )
  }

  render() {
    const {
      addon,
      type,
      placeholder,
      name,
      id,
      labelledBy,
      label,
      className,
      error,
      disabled,
      readOnly,
      onFocus,
      onBlur,
      autoFocus,
      required,
      maxLength,
      autoCorrect,
      autoComplete,
      title,
      buttonTitle,
      format,
      inputMode,
      pattern,
      focusBorder,
      loader,
      inputAttributes,
    } = this.props
    const value = this.state.value ? format(this.state.value, this.state.previousValue) : ''

    const attrs: InputHTMLAttributes<HTMLInputElement> & {
      ref: any
    } = {
      ...inputAttributes,
      type,
      placeholder,
      name,
      id,
      'aria-labelledby': labelledBy,
      value,
      maxLength,
      autoComplete,
      autoCorrect,
      title,
      inputMode,
      pattern,
      // modifiers
      disabled,
      readOnly,
      required,
      autoFocus,
      // actions
      onFocus,
      onBlur,
      onChange: this.onTextFieldChange,
      ref: this.ref,
    }

    if (error) {
      attrs['aria-invalid'] = 'true'
    }

    const iconProps = {
      iconColor: color.lightMidnightGreen,
      size: '18',
    }

    if (type === inputTypes.NUMBER) {
      // Display numeric keyboard on iOS
      attrs.pattern = '[0-9]*'
      attrs.inputMode = inputModes.NUMERIC
    }

    const buttonOnClick = type !== inputTypes.PASSWORD ? this.clearValue : this.toggleShowPassword
    const shouldDisplayButton = !disabled && value

    return (
      <StyledTextField
        className={cc(['kirk-textField', prefix({ error: !!error, disabled }), className])}
      >
        {label && <label htmlFor={id}>{label}</label>}
        <div
          className={cc([
            'kirk-textField-wrapper',
            {
              'kirk-textField-wrapper--hasFocus': focusBorder && this.state.hasFocus,
            },
          ])}
        >
          {addon}
          <input
            {...attrs}
            onFocus={this.onFocus}
            onBlur={this.onBlur}
            type={type === inputTypes.PASSWORD && this.state.showPassword ? inputTypes.TEXT : type}
          />
          {loader}
          {!loader && shouldDisplayButton && (
            <Button
              className="kirk-textField-button"
              status={ButtonStatus.UNSTYLED}
              isBubble
              onClick={buttonOnClick}
              tabIndex={-1}
              title={buttonTitle}
              aria-hidden={isEmpty(buttonTitle)}
              buttonRef={(elem: HTMLButtonElement) => {
                this.clearButton = elem
              }}
            >
              {type === 'password' ? (
                <EyeIcon {...iconProps} off={this.state.showPassword} />
              ) : (
                <CrossIcon {...iconProps} />
              )}
            </Button>
          )}
        </div>
        {Boolean(error) && this.renderError()}
      </StyledTextField>
    )
  }
}
