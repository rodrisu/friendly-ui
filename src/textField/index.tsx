export { TextField, TextFieldProps, inputModes, inputTypes } from './TextField'
