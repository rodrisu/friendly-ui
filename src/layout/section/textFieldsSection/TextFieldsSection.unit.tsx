import React from 'react'
import renderer from 'react-test-renderer'

import { TextField } from '../../../textField'
import { TextFieldsSection } from './index'

describe('TextFieldsSection', () => {
  it('should render default textField section', () => {
    const section = (
      <TextFieldsSection>
        <TextField
          name="firstInputSecondRow"
          placeholder="In-section input"
          onChange={() => null}
        />
        <TextField
          name="secondInputSecondRow"
          placeholder="In-section input"
          onChange={() => null}
        />
      </TextFieldsSection>
    )
    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })

  it('should render non-default textField section', () => {
    const props = {
      tagName: 'span',
      className: 'content',
    }
    const section = (
      <TextFieldsSection {...props}>
        <TextField
          name="firstInputSecondRow"
          placeholder="In-section input"
          onChange={() => null}
        />
        <TextField
          name="secondInputSecondRow"
          placeholder="In-section input"
          onChange={() => null}
        />
        <TextField
          name="thirdInputSecondRow"
          placeholder="In-section input"
          onChange={() => null}
        />
      </TextFieldsSection>
    )

    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })
})
