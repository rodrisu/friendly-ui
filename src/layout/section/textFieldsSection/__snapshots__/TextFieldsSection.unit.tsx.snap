// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldsSection should render default textField section 1`] = `
.c2 {
  position: relative;
  box-sizing: border-box;
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
  padding: 8px 0;
}

.c2 .kirk-textField-wrapper {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  color: #054752;
  background-color: #EDEDED;
  border-radius: 16px;
  border: solid 1px #EDEDED;
  box-shadow: none;
}

.c2 .kirk-textField-wrapper--hasFocus {
  border: 3px solid #00AFF5;
}

.c2 .kirk-textField-wrapper--hasFocus input,
.c2 .kirk-textField-wrapper--hasFocus textarea {
  padding: calc(16px + 1px - 3px);
}

.c2 .kirk-textField-wrapper--hasFocus input {
  padding-right: 0;
}

.c2 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  border-radius: 16px;
  background-color: #EDEDED;
  color: #054752;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  line-height: 20px;
  width: 100%;
  caret-color: #00AFF5;
  padding: 16px;
}

.c2 input {
  padding-right: 0;
  font-family: inherit;
  margin-right: 48px;
}

.c2 input:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  padding-left: 0;
}

.c2 input::-webkit-input-placeholder {
  color: #708C91;
}

.c2 input::-moz-placeholder {
  color: #708C91;
}

.c2 input:-ms-input-placeholder {
  color: #708C91;
}

.c2 input::placeholder {
  color: #708C91;
}

.c2 input::-ms-clear {
  display: none;
}

.c2 input[type='number']::-webkit-outer-spin-button,
.c2 input[type='number']::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

.c2 input[type='number'] {
  -moz-appearance: textfield;
  -webkit-appearance: none;
}

.c2 input[type='search'] {
  width: 100%;
  box-sizing: border-box;
}

.c2 input[type='number'],
.c2 input[type='search'] {
  box-shadow: none;
}

.c2 input:focus {
  border-color: #EDEDED;
  box-shadow: none;
  outline: none;
}

.c2.kirk-error .kirk-textField-wrapper {
  background: #FDD8DE;
  border: solid 1px #FDD8DE;
  -webkit-animation: textFieldError 500ms ease-in-out;
  animation: textFieldError 500ms ease-in-out;
}

.c2.kirk-error input {
  background-color: #FDD8DE;
}

.c2.kirk-error .kirk-error-message {
  color: #F53F5B;
  display: block;
  padding: 8px;
}

.c2 label {
  color: #054752;
  padding: 0 8px 4px 8px;
}

.c2 .kirk-button {
  background-color: transparent;
  color: #708C91;
  height: auto;
}

.c2 .kirk-textField-button {
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
}

.c2 .kirk-textField-button[hidden] {
  visibility: hidden;
}

.c2.kirk-disabled {
  opacity: 0.5;
  cursor: default;
}

.c0 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c0 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

.c1 .kirk-textfields-section-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 .kirk-textfields-section-content .kirk-textField {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 .kirk-textfields-section-content .kirk-textField:first-child {
  padding-right: 4px !important;
}

.c1 .kirk-textfields-section-content .kirk-textField:last-child {
  padding-left: 4px !important;
}

@media (min-width:800px) {
  .c0 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c0 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<div
  className="c0 c1"
  role="presentation"
>
  <div
    className="section-content kirk-textfields-section-content section-content--noHorizontalSpacing"
  >
    <div
      className="c2 kirk-textField"
    >
      <div
        className="kirk-textField-wrapper"
      >
        <input
          name="firstInputSecondRow"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="In-section input"
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="c2 kirk-textField"
    >
      <div
        className="kirk-textField-wrapper"
      >
        <input
          name="secondInputSecondRow"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="In-section input"
          type="text"
          value=""
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`TextFieldsSection should render non-default textField section 1`] = `
.c2 {
  position: relative;
  box-sizing: border-box;
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
  padding: 8px 0;
}

.c2 .kirk-textField-wrapper {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  color: #054752;
  background-color: #EDEDED;
  border-radius: 16px;
  border: solid 1px #EDEDED;
  box-shadow: none;
}

.c2 .kirk-textField-wrapper--hasFocus {
  border: 3px solid #00AFF5;
}

.c2 .kirk-textField-wrapper--hasFocus input,
.c2 .kirk-textField-wrapper--hasFocus textarea {
  padding: calc(16px + 1px - 3px);
}

.c2 .kirk-textField-wrapper--hasFocus input {
  padding-right: 0;
}

.c2 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  border-radius: 16px;
  background-color: #EDEDED;
  color: #054752;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  line-height: 20px;
  width: 100%;
  caret-color: #00AFF5;
  padding: 16px;
}

.c2 input {
  padding-right: 0;
  font-family: inherit;
  margin-right: 48px;
}

.c2 input:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  padding-left: 0;
}

.c2 input::-webkit-input-placeholder {
  color: #708C91;
}

.c2 input::-moz-placeholder {
  color: #708C91;
}

.c2 input:-ms-input-placeholder {
  color: #708C91;
}

.c2 input::placeholder {
  color: #708C91;
}

.c2 input::-ms-clear {
  display: none;
}

.c2 input[type='number']::-webkit-outer-spin-button,
.c2 input[type='number']::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

.c2 input[type='number'] {
  -moz-appearance: textfield;
  -webkit-appearance: none;
}

.c2 input[type='search'] {
  width: 100%;
  box-sizing: border-box;
}

.c2 input[type='number'],
.c2 input[type='search'] {
  box-shadow: none;
}

.c2 input:focus {
  border-color: #EDEDED;
  box-shadow: none;
  outline: none;
}

.c2.kirk-error .kirk-textField-wrapper {
  background: #FDD8DE;
  border: solid 1px #FDD8DE;
  -webkit-animation: textFieldError 500ms ease-in-out;
  animation: textFieldError 500ms ease-in-out;
}

.c2.kirk-error input {
  background-color: #FDD8DE;
}

.c2.kirk-error .kirk-error-message {
  color: #F53F5B;
  display: block;
  padding: 8px;
}

.c2 label {
  color: #054752;
  padding: 0 8px 4px 8px;
}

.c2 .kirk-button {
  background-color: transparent;
  color: #708C91;
  height: auto;
}

.c2 .kirk-textField-button {
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
}

.c2 .kirk-textField-button[hidden] {
  visibility: hidden;
}

.c2.kirk-disabled {
  opacity: 0.5;
  cursor: default;
}

.c0 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c0 .section-content--noHorizontalSpacing {
  padding-left: 0;
  padding-right: 0;
}

.c1 .kirk-textfields-section-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 .kirk-textfields-section-content .kirk-textField {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 .kirk-textfields-section-content .kirk-textField:first-child {
  padding-right: 4px !important;
}

.c1 .kirk-textfields-section-content .kirk-textField:last-child {
  padding-left: 4px !important;
}

@media (min-width:800px) {
  .c0 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c0 .section-content.section-content--large {
    max-width: 1016px;
  }
}

<div
  className="c0 c1 content"
  role="presentation"
>
  <span
    className="section-content kirk-textfields-section-content section-content--noHorizontalSpacing"
  >
    <div
      className="c2 kirk-textField"
    >
      <div
        className="kirk-textField-wrapper"
      >
        <input
          name="firstInputSecondRow"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="In-section input"
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="c2 kirk-textField"
    >
      <div
        className="kirk-textField-wrapper"
      >
        <input
          name="secondInputSecondRow"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="In-section input"
          type="text"
          value=""
        />
        
      </div>
    </div>
    <div
      className="c2 kirk-textField"
    >
      <div
        className="kirk-textField-wrapper"
      >
        <input
          name="thirdInputSecondRow"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="In-section input"
          type="text"
          value=""
        />
        
      </div>
    </div>
  </span>
</div>
`;
